### PR TITLE
Add n21527

### PR DESCRIPTION
**Pull request by jnewbery**

### CODE REVIEW

File: _posts/2021-06-02-#21527.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  the code will behave.
+
+- This PR does two things:
+
+  - introduces an `m_mutex_message_handling` mutex in `PeerManager`, which guards data
+    that is used in `PeerManager`'s message handling. Doing that allows us to remove
+    the `cs_sendProcessing` mutex from net.
+
+  - removes the `g_cs_orphans` mutex from net_processing and adds an internal
+    mutex to `TxOrphange`, making it responsible for its own locking and thread
```
Comment (jonatack) [inline _posts/2021-06-02-#21527.md:35]:
```suggestion
    mutex to `TxOrphanage`, making it responsible for its own locking and thread
```

File: _posts/2021-06-02-#21527.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+   What was your review approach?
+
+2. Does this PR change observable behaviour of the node in any way?
+
+3. What was the `cs_sendProcessing` mutex responsible for before this PR? When
+   was `cs_sendProcessing` introduced? (hint use `git blame` and `git log -S
```
Comment (jonatack) [inline _posts/2021-06-02-#21527.md:47]:
```suggestion
   was `cs_sendProcessing` introduced? (Hint: use `git blame` and `git log -S
```

File: _posts/2021-06-02-#21527.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   "string"` to search back through previous commits).
+
+4. This PR moves one mutex from net to net_processing
+   (`CNode.cs_sendProcessing` is replaced by
+   `PeerManager.m_mutex_message_handler`) and one mutex from net_processing to
+   txorphange (`g_cs_orphans` is replaced by `TxOrphanage.m_mutex`). What are
```
Comment (jonatack) [inline _posts/2021-06-02-#21527.md:53]:
```suggestion
   txorphanage (`g_cs_orphans` is replaced by `TxOrphanage.m_mutex`). What are
```

File: _posts/2021-06-02-#21527.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+---
+layout: pr
+date: 2021-06-02
+title: "Net_processing: lock clean up"
+pr: 21527
+authors: [ajtowns]
+components: ["p2p", "refactoring"]
+host: jnewbery
+status: upcoming
+commit:
```
Comment (glozow) [inline _posts/2021-06-02-#21527.md:10]: needs a commit?

File: _posts/2021-06-02-#21527.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+5. What are `vExtraTxnForCompact` and `vExtraTxnForCompactIt`? Why is it ok to
+   stop guarding them with `g_cs_orphans` and guard them with
+   `m_mutex_message_handling` instead?
+
+6. This PR removes the `Peer.m_orphan_work_set` member and replaces it with
+   a `m_peer_work_set` map in `TxOrphanage`. What is the peer orphan work set
+   used for?
+
+7. (bonus question) This PR originally included some behavioural changes in the
```
Comment (glozow) [inline _posts/2021-06-02-#21527.md:65]:
```suggestion
7. (Bonus question) This PR originally included some behavioural changes in the
```

File: _posts/2021-06-02-#21527.md (hunk 4 above)
Comment (jonatack) [reply _posts/2021-06-02-#21527.md:10]: The commit hash isn't used until the status changes to "past" (so it is often done when doing that and adding the meeting log). Also, the PR author might re-push during the days before the meeting and (sometimes) it makes sense to use the latest one.

File: _posts/2021-06-02-#21527.md (hunk 3 above)
Comment (jnewbery) [reply _posts/2021-06-02-#21527.md:53]: fixed

File: _posts/2021-06-02-#21527.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+   way that orphans were handled. Those have now been moved from this PR to
+   [a separate branch](https://github.com/ajtowns/bitcoin/commits/202104-whohandlesorphans),
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/2021-06-02-#21527.md:67]: done

File: _posts/2021-06-02-#21527.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+5. What are `vExtraTxnForCompact` and `vExtraTxnForCompactIt`? Why is it ok to
+   stop guarding them with `g_cs_orphans` and guard them with
+   `m_mutex_message_handling` instead?
+
+6. This PR removes the `Peer.m_orphan_work_set` member and replaces it with
+   a `m_peer_work_set` map in `TxOrphanage`. What is the peer orphan work set
+   used for?
+
+7. (bonus question) This PR originally included some behavioural changes in the
+   way that orphans were handled. Those have now been moved from this PR to
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/2021-06-02-#21527.md:66]: fixed

File: _posts/2021-06-02-#21527.md (hunk 5 above)
Comment (jnewbery) [reply _posts/2021-06-02-#21527.md:65]: done!

**Comment by jnewbery** [comment]
Thanks for the reviews @glozow and @jonatack. All review comments addressed.

**Pull request merged**